config/spapi.php

```diff
@@ -25,4 +20,15 @@
         // Valid options are NA, EU, FE
         'endpoint' => env('SPAPI_ENDPOINT_REGION', 'NA'),
     ],
+
+    /**
+     * Can be used as a security switch to turn API class registration
+     * completely off for non-production environments.
+     * If enabled, obtaining instances of API classes might cause
+     * unintended API requests using the configured credentials.
```

Comment:
Could you clarify that these would be authorization-related requests to Amazon's OAuth servers, not SP API requests? Just want to make sure people don't think we're making SP API requests in the background with their credentials.

Reply:
I changed the comment a little, but I also don't want to go too much into the details. If you have an exact wording you'd like to see, please share.

On the other hand and the more I'm thinking about it...
If it's too complicated to explain why one would need this switch, we should probably just leave it out. 

Before changing the binding, it was really a no-go where each tests just potentially fired auth requests directly when the service provider was loaded. But now since this does not happen anymore, I guess it's OK to leave it to the developers to care that there are no requests made during the tests. 

The only thing it solves for me right now, is that I can use production credentials in my local testing setup and just deactivate using them with this switch. Then I just mock the request for normal test runs, and when I want to test with real requests I just activate this switch, remove the mocking parts and run the tests with the real credentials.

But I don't really see that something like this is enough to include this in the library.

Reply:
Ok, yes, I agree. The use case for this feels small enough that I'm not sure it's worth the added config complexity for the end user. People can figure out their own prod/test credentials setups themselves, I think.